heapster-saw/src/Verifier/SAW/Heapster/Implication.hs

```diff
@@ -7988,6 +8004,8 @@ isProvablePerm unsetVars maybe_x p
   | neededs <- maybe id (\x -> NameSet.insert x) maybe_x $ neededVars p
```

Comment:
This line adds `x` to the needed variables, but it is not needed after this change if it has unit type. So maybe there should be a case that checks that? The difficulty is then that `isProvablePerm` needs to take in the type of each variable as well, i.e., needs an input argument of type `Maybe (TypeRepr a, ExprVar a)` instead of just `Maybe (ExprVar a)`. (Bundling the type with the `ExprVar` inside the `Maybe` allows the caller to avoid giving the type when the variable is not supplied.)